whirlpool/auth.py

```diff
@@ -57,7 +58,7 @@ def _save_auth_data(self):
         with open(AUTH_JSON_FILE, "w") as f:
             json.dump(self._auth_dict, f)
 
-    async def _do_auth(self, refresh_token):
+    async def _do_auth(self, refresh_token: str) -> str | None:
```

Comment:
This is an example - if you don't have `from __future__ import annotations` this will not work.

Reply:
I have no issues here on that line

Reply:
This fails the Python 3.9 tests without it...